FaunaDB.Client/Utils/CheckLatestVersion.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace FaunaDB.Client.Utils
+{
+    public class CheckLatestVersion
+    {
+        private const string packageName = "FaunaDB.Client";
+        public static bool? AlreadyChecked { get; set; } = null;
+        public static async Task GetVersionAsync()
+        {
+            if (AlreadyChecked.HasValue) return;
+            var latestNuGetVesrionString = string.Empty;
+            var url = $"https://api.nuget.org/v3-flatcontainer/{packageName}/index.json";
+            try
+            {
+                var httpClient = new HttpClient();
+#if NET45
+                ServicePointManager.SecurityProtocol = SecurityProtocolType.Ssl3 | SecurityProtocolType.Tls | SecurityProtocolType.Tls11 | SecurityProtocolType.Tls12;
+#endif
+                var response = await httpClient.GetAsync(url);
+                string versionsResponse = await response.Content.ReadAsStringAsync();
+                Newtonsoft.Json.Linq.JObject jObject = Newtonsoft.Json.Linq.JObject.Parse(versionsResponse);
+                var latestNuGetVesrion = ((Newtonsoft.Json.Linq.JArray)jObject.First.First).Children().LastOrDefault();
+                latestNuGetVesrionString = latestNuGetVesrion.ToString();
+                AlreadyChecked = true;
+            }
+            catch(Exception ex)
+            {
+                AlreadyChecked = false;
+                var message = $"Enable to check new Fauna driver version. Exception: {ex.Message}";
+                return;
+            }
+
+            Assembly asm = typeof(CheckLatestVersion).GetTypeInfo().Assembly;
+            var currentVersion =  asm.GetName().Version;
+            var currentVersionString = $"{currentVersion.Major}.{currentVersion.Minor}.{currentVersion.Build}";
+            if (!latestNuGetVesrionString.Equals(currentVersionString))
+            {
+                var message =  GetMessage(latestNuGetVesrionString, currentVersionString);
+                Debug.WriteLine(message);
```

Comment:
Is it a good practice to leave `Debug` messages in the production code in c#?

Reply:
Sure. It works only in Debug mode. In release mode it doesn't work